terraform/Makefile

```diff
@@ -30,6 +30,18 @@ clean: tfclean
 build: clean
 	GOOS=$(OS) GOARCH=$(ARCH) $(CGOFLAG) go build -o $(BUILDDIR)/terraform-provider-teleport $(BUILDFLAGS)
 
+build-darwin-universal: $(addprefix $(BUILDDIR)/terraform-provider-teleport_,arm64 amd64)
+	lipo -create -output $(BUILDDIR)/terraform-provider-teleport $^
+	rm $^
+
+# Architecture-specific binaries for the universal binary are extracted from
+# the release tarball. make will not automatically build this; you will need
```

Comment:
Will this at least fail if you forget to build the release tarball first?

Reply:
Yes. Make will say:
```
make[1]: *** No rule to make target 'build/terraform-provider-teleport_arm64', needed by 'build-darwin-universal'.  Stop.
```

I did try to add a rule to build it automatically, but due to ordering and that `release` does a clean which removes the `build` directory, it ends up deleting the first binary when it builds the second, and I didn't want to get into refactoring the `clean` targets.